workflows/sars-cov-2-variant-calling/sars-cov-2-pe-illumina-artic-ivar-analysis/pe-wgs-ivar-analysis.ga

```diff
@@ -0,0 +1,1215 @@
+                "top": 326.33331298828125,
+                "width": 200,
+                "x": 474.75,
+                "y": 326.33331298828125
+            },
+            "post_job_actions": {
+                "HideDatasetActionconsensus": {
+                    "action_arguments": {},
+                    "action_type": "HideDatasetAction",
+                    "output_name": "consensus"
+                }
+            },
+            "tool_id": "toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.3.1+galaxy0",
+            "tool_shed_repository": {
+                "changeset_revision": "3bc2ef3d4a17",
+                "name": "ivar_consensus",
+                "owner": "iuc",
+                "tool_shed": "toolshed.g2.bx.psu.edu"
+            },
+            "tool_state": "{\"filter_depth\": \"false\", \"gap\": \"true\", \"input_bam\": {\"__class__\": \"ConnectedValue\"}, \"min_depth\": \"50\", \"min_freq\": {\"__class__\": \"ConnectedValue\"}, \"min_qual\": {\"__class__\": \"ConnectedValue\"}, \"__page__\": null, \"__rerun_remap_job_id__\": null}",
+            "tool_version": "1.3.1+galaxy0",
+            "type": "tool",
+            "uuid": "5fb477e6-e7ac-4e02-a31f-335228becdf3",
+            "workflow_outputs": []
+        },
+        "13": {
+            "annotation": "",
+            "content_id": "__FLATTEN__",
+            "errors": null,
+            "id": 13,
+            "input_connections": {
+                "input": {
+                    "id": 10,
+                    "output_name": "raw_data"
+                }
+            },
+            "inputs": [],
+            "label": null,
+            "name": "Flatten Collection",
+            "outputs": [
+                {
+                    "name": "output",
+                    "type": "input"
+                }
+            ],
+            "position": {
+                "bottom": -122.06666564941406,
+                "height": 113.60000610351562,
+                "left": 474.75,
+                "right": 674.75,
+                "top": -235.6666717529297,
+                "width": 200,
+                "x": 474.75,
+                "y": -235.6666717529297
+            },
+            "post_job_actions": {
+                "HideDatasetActionoutput": {
+                    "action_arguments": {},
+                    "action_type": "HideDatasetAction",
+                    "output_name": "output"
+                }
+            },
+            "tool_id": "__FLATTEN__",
+            "tool_state": "{\"input\": {\"__class__\": \"ConnectedValue\"}, \"join_identifier\": \"_\", \"__page__\": null, \"__rerun_remap_job_id__\": null}",
+            "tool_version": "1.0.0",
+            "type": "tool",
+            "uuid": "f2fe26ec-f30b-4bc7-9dd6-62f3c9538ae7",
+            "workflow_outputs": []
+        },
+        "14": {
+            "annotation": "",
+            "content_id": "toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1",
```

Comment:
This step seems *very* questionable!

Just renaming the reference to have it recognized by SnpEff can mask input data issues. What if the reference used by the user isn't identical to NC_045512.2?

Reply:
My assumption is you do this to keep both ivar trim and SnpEff happy?
If that's the case you can just omit this step and directly use NC_045512.2 as input. ivar trim will change the primer bed file ref name on the fly for you.

Reply:
The renaming is because MN908947.3 is often used as input and yes it does have a risk like you outlined.

Reply:
Ok, then what about this compromise: You replace specifically only MN908947.3 with NC_045512.2, but do this up front on the user-provided fasta file?

Those two are base identical, correct?

Reply:
Should we not explicitly state which input we assume, which reference genome, and not hack around potential user assumptions. This is a general question for all workflows under IWC.

Reply:
In general, I agree (need to check our existing WFs 😊 ). In the specific case though MN908947.3 is the ID of the record the refseq NC_045512.2 was generated from and the two are identical and used kind of interchangeably (the ARTIC networks primer scheme files, e.g., use MN908947.3 instead of the refseq ID). They are identical (I just reconfirmed) so there's a point to making the WF convenient to use.
I don't think I will add such an extra step to the covid19.galaxyproject.org WFs, but this one here really emphasizes end-user friendliness so maybe it's an appropriate thing to do here.
As I said, I would like to see the replacement done as the very first thing in the WF though. No changing of IDs half-way through the history.